pkg/apis/elasticsearch/v1alpha1/types.go

```diff
@@ -107,6 +107,7 @@ const (
 // ElasticsearchStatus represents the status of Elasticsearch cluster
 type ElasticsearchStatus struct {
 	// Fill me
-	Nodes    []ElasticsearchNodeStatus   `json:"nodes"`
-	K8sState ElasticsearchRequiredAction `json:"clusterState"`
+	Nodes         []ElasticsearchNodeStatus   `json:"nodes"`
+	K8sState      ElasticsearchRequiredAction `json:"clusterState"`
```

Comment:
I realize this was here before but why is the field 'K8sState'?  Looking at the actions, they don't appear in anyway specifically tied to kubernetes.  We should consider re-naming this, IMO, to something like `ClusterState` or even more descriptive of `requiredAction`.  Calling it even `clusterState` seems misleading as `state` implies a noun (or maybe a verb? e.g. 'restarting', 'ok') to me or a place in time where the value describes an action that needs to occur.  @ewolinetz thoughts?

Reply:
1) this is still work in progress
2) I'm thinking about removing it altogether - what is the value to a sysadmin what was the last action taken by the operator?

Reply:
Based on the available actions  it looks like we are trying to convey the ES cluster has to be restarted in some specific way though the way they are written it implies to me that someone should do something to resolve the state.  I think they would be better written in present tense defining what the operator is doing to the cluster(e.g. performingRollingRestart) or the state its in (e.g. restarting).  Is there value in knowing its a full restart vs a rolling restart?  Maybe we need a state transition diagram to understand what state the cluster could be in and where it could go from one state to the next.  From a user perspective that makes more sense to define state because I get the ES type and am able to see what is happening to the cluster: the operator is performing a rolling restart.

Reply:
I vote to remove it. It seems like this is part of where we had an overlap of what status should mean and here it is being used by the operator only. I think this would be better served with a "Message" field that described a list of the things the operator had last changed/done/performed since the last CR change.